### PR TITLE
feat(agent): verify RFC-64 catalog row authorship

### PR DESCRIPTION
**Pull request by branarakic**
Implements the frozen OT-RFC-64 §4.4.3 catalog-row authorship boundary as an opaque process-local capability. It closes direct/delegated author authority, the deployed v10 agent-delegation proof, exact control-envelope signature capabilities, head/directory/bucket/row/transfer binding, half-open validity intervals, and packed author identity without claiming policy, timeliness, chain ownership, projection validity, or semantic admission.

Security hardening includes canonical snapshots, unforgeable WeakMap tokens, closed errors, exact target-digest assertions, and an up-front directoryHeight + 1 path bound before any attacker-controlled element is parsed or signature-bound.

Validation:
- focused RFC-64 authorship: 30/30
- complete agent unit suite: 1,226 passed, 4 skipped
- agent build and type tests
- git diff check

Spec: OriginTrail/dkgv10-spec#144
Integration target: integration/rfc64-devnet

### CODE REVIEW

File: packages/agent/src/rfc64/catalog-row-authorship.ts [hunk 1]
```diff
@@ -0,0 +1,1145 @@
...
+  const snapshot = snapshotParentEvidence(evidence);
+  return computeParentEvidenceDigestAfterSnapshot(snapshot);
+}
+
+/**
+ * Close one exact signed catalog row under its author-authorized catalog key.
+ * No policy, timeliness, history, finality, persistence, RDF, or activation work
+ * is performed here.
+ */
+export function verifyAuthorCatalogRowAuthorshipV1(
```
Comment (otReviewAgent) [inline packages/agent/src/rfc64/catalog-row-authorship.ts:242]: 🟡 Issue: This lands as an oversized all-in-one verifier module

**What's wrong**
The PR introduces a new production file above the 1k-line threshold without a clear structural reason. The code has several separable responsibilities, but they are packed into one module, which makes future RFC-64 changes more likely to grow by appending more branches and helper code instead of extending the right abstraction.

**Example**
A reader trying to understand `verifyAuthorCatalogRowAuthorshipV1` has to scan from the public verifier at line 242 through helper blocks all the way to line 1145, with no module boundary separating authority resolution, catalog object binding, and low-level input hardening.

**Suggested direction**
Decompose this into a small public `index.ts` plus focused implementation files. A good split would make the top-level verifier read as orchestration over named concepts rather than embedding every parser, proof binder, and primitive validator in one file.

**For Agents**
Split `packages/agent/src/rfc64/catalog-row-authorship.ts` before merging. Preserve exported behavior, but move parent-authority/evidence parsing, capability registry, directory/bucket/row binding, and shared snapshot utilities into focused modules with a small barrel export. Existing tests should continue to exercise the same public entry points.

File: packages/agent/src/rfc64/catalog-row-authorship.ts [hunk 2]
```diff
@@ -0,0 +1,1145 @@
...
+    fail(
+      'AUTHORSHIP_INTERVAL_MISMATCH',
+      'catalog head issuedAt is outside the child half-open interval',
+    );
+  }
+
+  const scope = deriveAuthorCatalogScopeFromHeadV1(head.payload);
+  const catalogScopeDigest = computeAuthorCatalogScopeDigestV1(scope);
+  const selectedBucketId = catalogKeyToBucketIdV1(targetKaId, scope.bucketCount);
+  const descriptor = readBoundPathDescriptor(
```
Comment (otReviewAgent) [inline packages/agent/src/rfc64/catalog-row-authorship.ts:414]: 🟡 Issue: The path proof API carries two sources of truth for the same closure

**What's wrong**
The current structure is more complicated than necessary: `directoryPathProof` is a process-local capability, but the verifier cannot trust it alone because it still needs the raw path for node signatures, so it recomputes the proof anyway. This is avoidable orchestration complexity and exposes callers to an extra state-sync obligation.

**Example**
Every caller must pre-mint a `VerifiedAuthorCatalogDirectoryPathV1` and also pass the same raw path envelopes. If those two representations drift, the verifier has to diagnose a mismatch that would not exist if the verifier owned the path proof computation.

**Suggested direction**
Make the authorship verifier accept raw path artifacts only and mint/read the directory-path capability internally. That deletes the reconciliation helper and makes the proof boundary atomic.

**For Agents**
In `VerifyAuthorCatalogRowAuthorshipInputV1`, remove `directoryPathProof` unless there is a separate caller-owned trust boundary not visible here. In `verifyAuthorCatalogRowAuthorshipV1`, compute `const pathProof = verifyAuthorCatalogDirectoryPathV1(head, directoryEnvelopes, selectedBucketId)` and read the descriptor from that single proof. Update fixtures to stop precomputing and passing `directoryPathProof`; preserve all current successful and failure behavior around invalid raw paths/signatures.

File: packages/agent/src/rfc64/catalog-row-authorship.ts [hunk 3]
```diff
@@ -0,0 +1,1145 @@
...
+    governanceContractAddress: delegationPayload.governanceContractAddress,
+    networkId: delegationPayload.networkId,
+    ownershipTransitionDigest: delegationPayload.ownershipTransitionDigest,
+    subGraphName: delegationPayload.subGraphName,
+  });
+  const authorCatalogAgentScopeDigest = computeAuthorCatalogAgentScopeDigestV1(agentScope);
+  const expectedAgentScope =
+    `${AUTHOR_CATALOG_AGENT_SCOPE_PREFIX_V1}${authorCatalogAgentScopeDigest.slice(2)}`;
+
+  const issuerIsAuthor = delegation.issuer === delegationPayload.authorAddress;
```
Comment (otReviewAgent) [inline packages/agent/src/rfc64/catalog-row-authorship.ts:320]: 🟡 Issue: Direct/delegated authority resolution is encoded as scattered branch state

**What's wrong**
The top-level verifier is carrying the direct-vs-delegated state machine inline via booleans and nullable fields. That makes the main flow harder to scan and makes future authority variants likely to add more special-case checks in the middle of unrelated catalog binding logic.

**Example**
To determine which catalog issuer key is trusted, a reader must follow the branch at lines 322-378 and then remember the additional `!issuerIsAuthor` condition at line 380 before the verifier moves on to head, bucket, and row binding.

**Suggested direction**
Reframe this as a small typed authority model or resolver so direct and delegated issuance become isolated cases and the row verifier remains a linear catalog-binding pipeline.

**Confidence note**
This is a structural concern; the current logic may be behaviorally correct, but the maintainability cost is visible from the diff.

**For Agents**
Extract an authority resolver, for example `resolveCatalogIssuerAuthorityV1(delegation, parentEvidence)`, that returns a frozen normalized authority snapshot with `authorAddress`, `catalogIssuerKey`, `effectiveAt`, `expiresAt`, `evidenceDigest`, and `agentScopeDigest`. Keep direct and delegated-specific validation inside that resolver, then let the main verifier consume only the normalized authority.

File: packages/agent/src/rfc64/catalog-row-authorship.ts [hunk 4]
```diff
@@ -0,0 +1,1145 @@
...
+): VerifiedAuthorCatalogRowAuthorshipV1 {
+  const immutable = deepFreezeJson({ ...snapshot }) as VerifiedAuthorCatalogRowAuthorshipSnapshotV1;
+  const capability = Object.freeze(
+    Object.create(null),
+  ) as VerifiedAuthorCatalogRowAuthorshipV1;
+  VERIFIED_AUTHOR_CATALOG_ROW_AUTHORSHIPS_V1.set(capability as object, immutable);
+  return capability;
+}
+
+function exactPrimitiveDataSnapshot(
```
Comment (otReviewAgent) [inline packages/agent/src/rfc64/catalog-row-authorship.ts:953]: 🟡 Issue: The snapshot helpers erase types and duplicate canonical-object plumbing

**What's wrong**
This adds a cast-heavy boundary around the most important verifier inputs. The generic helper makes the code look reusable, but it pushes the real invariants into scattered assertions and final casts, while also duplicating canonical object handling that already has a home in the RFC-64/core layer.

**Example**
`snapshotAuthorCatalogAgentScope` validates each field, but the helper erases the shape first, so the function has to recover type information by assertion and final cast instead of returning a typed `AuthorCatalogAgentScopeV1` from a dedicated parser.

**Suggested direction**
Move from generic `Record<string, unknown>` snapshots to typed parse/normalize functions, preferably built on the same canonical helpers used by the existing RFC-64 codecs.

**For Agents**
Replace `exactPrimitiveDataSnapshot` plus casts with typed parsers/builders for `AuthorCatalogAgentScopeV1` and `AuthorAgentDelegationEvidenceV1`. Reuse or promote the existing RFC-64 exact-key/canonical JSON helpers from core where appropriate, and keep only authorship-specific validation in this module.

File: packages/agent/src/rfc64/catalog-row-authorship.ts [hunk 5]
```diff
@@ -0,0 +1,1145 @@
...
+  ) {
+    fail(
+      'AUTHORSHIP_INTERVAL_MISMATCH',
+      'catalog head issuedAt is outside the child half-open interval',
+    );
+  }
+
+  const scope = deriveAuthorCatalogScopeFromHeadV1(head.payload);
+  const catalogScopeDigest = computeAuthorCatalogScopeDigestV1(scope);
+  const selectedBucketId = catalogKeyToBucketIdV1(targetKaId, scope.bucketCount);
```
Comment (otReviewAgent) [inline packages/agent/src/rfc64/catalog-row-authorship.ts:413]: 🟡 Issue: Single-bucket fixtures leave bucket selection and multi-node path closure unverified

**What's wrong**
The changed verifier’s bucket-selection and path-binding behavior is security-relevant, but the added tests never leave the degenerate one-bucket, height-zero case. That gives confidence for the happy-path closure shape while leaving the non-trivial directory behavior untested.

**Example**
A regression such as replacing `catalogKeyToBucketIdV1(targetKaId, scope.bucketCount)` with a constant `'0'`, or accidentally ignoring intermediate directory nodes, would still pass the current authorship tests because no fixture has multiple buckets or a path longer than one node.

**Suggested direction**
Add at least one positive and one negative authorship test using a real multi-bucket catalog path, preferably with a non-zero selected bucket.

**For Agents**
In `packages/agent/test/rfc64-catalog-row-authorship.test.ts`, add a fixture with multiple buckets and `directoryHeight` greater than `0`, where the target `kaId` maps to a non-zero bucket. Prove the verifier accepts the valid multi-node path and rejects a bucket/path for a different selected bucket or a missing/wrong intermediate node.

**Comment by branarakic** [comment]
Closing — this PR's code is already on `main`, so merging it would change nothing.

**Evidence** (verified 2026-08-06 against `main` @ `0277d82c6`):
- `git cherry origin/main` reports **no new patches** — every content commit is already upstream; only 2 merge commits remain. The feature landed on main as `a456de2de` "feat(agent): verify RFC-64 catalog row authorship".
- `git merge-tree --write-tree origin/main 7bba6eb2a` returns main's exact tree (`d1812f293`) — a merge would be a no-op.

Closed as part of a sweep of open PRs whose code had already landed. Please reopen if this is wrong.
